### ISSUE TITLE
My OpenAI API not work after installation.

### ISSUE THREAD

**Issue by p6002** [reworkd/AgentGPT#609]
### Discussed in https://github.com/reworkd/AgentGPT/discussions/605

<div type='discussions-op-text'>

<sup>Originally posted by **p6002** May 26, 2023</sup>
I cloned the github repo, ran `./setup.sh --docker` API during installation shows Valid. 
When I log in via `localhost:3000` and run the task, it shows: ERROR when connecting to the OpenAI API key. Please check the API key or try later.
I checked and in next/.env there is a valid API on OpenAI. 
What am I doing wrong?

<img width="925" alt="image" src="https://github.com/reworkd/AgentGPT/assets/74392057/6d82089b-8643-4c0c-a40d-4036d51801d7">

<img width="699" alt="image" src="https://github.com/reworkd/AgentGPT/assets/74392057/5d790792-adeb-48c8-98e6-9219c18f6fc2">

<img width="394" alt="image" src="https://github.com/reworkd/AgentGPT/assets/74392057/ecbe36d1-45f7-4246-b99c-4b3595030a7e">

<img width="554" alt="image" src="https://github.com/reworkd/AgentGPT/assets/74392057/f664f91f-83d4-4261-bfbe-0b8b8bbb8ed5">

</div>

**Comment by asim-shrestha** [comment]
Hey, can you retry running the `./setup.sh` script? We've recently updated it

**Comment by p6002** [comment]
Previously I used `./setup.sh --docker`, now after using `./setup.sh` it shows:

![image](https://github.com/reworkd/AgentGPT/assets/74392057/faa69824-d886-4f29-8370-956b76e53c5e)

**Comment by asim-shrestha** [comment]
So you need to first fetch and pull the latest changes from master. To confirm this, make sure there is a `cli` directory.

Next, the error message shows you have an outdated version of node installed. If you have `nvm`

```
nvm install 18
nvm use 18
```

From there, re-run `setup.sh` it will ask you how you're running it (docker-compose, docker, etc) and ask you to provide a set of API keys for the different services we utilize

**Comment by p6002** [comment]
I did it and managed (after a few hours) to launch the new version. 
 (I had to manually add the server IP address instead of "localhost" in the .env files.

However, it did not change anything, and the error with the API still appears. 

The API is entered correctly during installation, the same API works in other tools without any problem.

**Comment by yiyunfei-001** [comment]
i also meet the problem,dose someone have solve the problem

**Comment by awtkns** [comment]
> I managed (after several hours) to get the new version working. (It was necessary to manually add the IP address of the server instead of "localhost" in the .env files.
> 
> However, it did not change anything, and the error with the API on the website still appears.
> 
> The API is entered correctly during installation (serper also), the same OpenAI API works in other tools without any problem.
> 
> This output generates the terminal while the application is running (after ./setup.sh): ![image](https://user-images.githubusercontent.com/74392057/241571006-fcb3fb03-3742-4420-8ae7-456b2a6da92b.png)

@p6002 given the error message you've provided it seems you may have something else running on port 8000? Could you be running this or another project locally?

**Comment by bitsnaps** [comment]
Trying different ways (with `setup.sh` or compile everyting manually npm, poetry...), I'm still getting `ERROR accessing OpenAI's API. Please check your API key or try again later`, the api key is working locally.

**Comment by JumpyWizard404** [comment]
Same issue when using the container version of the deployment

**Comment by JumpyWizard404** [comment]
Managed to get it to work: this is for the docker deployment one where i noticed that it was not properly formating the request as the environment variable was set to REWORKD_PLATFORM_OPENAI_API_BASE="https://api.openai.com/v1" the docker logs were showing that the URL it was sending requests to ended up being formated as "https://api.openai.com/v1"/chat/completions and were not communicating
After updating the environment variable to REWORKD_PLATFORM_OPENAI_API_BASE=https://api.openai.com/v1 it started proper communicating to the API, also note i used the version 5.0-beta from https://github.com/reworkd/AgentGPT/releases/tag/v.0.5.0-beta

**Comment by p6002** [comment]
After completely deleting and cloning the repo again, I was able to get the agent up and running. 
The only changes I made in `next/.env` instead of localhost I entered my server IP address here:
```
NEXT_PUBLIC_BACKEND_URL=
NEXTAUTH_URL=
REWORKD_PLATFORM_FRONTEND_URL=
```

**Comment by MSYXing** [comment]
May I ask how you started it? Don't you have the problem about "blocked by CORS policy"?
I also used the 0.6.0 version and completed the relevant changes you mentioned.
Here is my problom：
I tried running this command (docker-compose -f deploy/docker-compose.yml --project-directory . up --build) and it showed that it was started, but the request (http://localhost:3000/api/agent/start) showed "blocked by CORS policy", and then I added this link(https://cors-anywhere.herokuapp.com/http://localhost:3000/api/agent/start) to fix it, but now 403
